bundles/org.openhab.binding.boschshc/src/main/feature/feature.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<features name="org.openhab.binding.boschshc-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.4.0">
+	<repository>mvn:org.openhab.core.features.karaf/org.openhab.core.features.karaf.openhab-core/${ohc.version}/xml/features</repository>
+
+	<feature name="openhab-binding-boschshc" description="BoschSHC Binding" version="${project.version}">
+		<feature>openhab-runtime-base</feature>
+		<bundle start-level="80">mvn:org.openhab.addons.bundles/org.openhab.binding.boschshc/${project.version}</bundle>
```

Comment:
After switch the dependencies to `provided` you would then need to update your `feature.xml` to let the osgi system know where it can find those dependencies. This should solve the error your are seeing in the build.
```suggestion
		<feature>openhab-runtime-base</feature>
		<bundle dependency="true">mvn:org.bouncycastle/bcpkix-jdk15on/1.52</bundle>
		<bundle dependency="true">mvn:org.bouncycastle/bcprov-jdk15on/1.52</bundle>
		<bundle start-level="80">mvn:org.openhab.addons.bundles/org.openhab.binding.boschshc/${project.version}</bundle>
```

Reply:
Thanks a lot @cpmeister! Your suggestion solves the error on my machine.

This means that the maven ``pom.xml`` contains the "compile time" dependencies and the ``feature.xml`` defines the runtime OSGi environment.

I will tests the new binding jar - now only 88Kb big - and commit the changes afterwards.

Reply:
88kb?! 🎆

Reply:
Oh no 😧. During runtime I get now problems: Not all used class from ``org.bouncycastle.*`` are contained in ``org.openhab.io.jetty.certificate`` and this binding code uses of cause missing classes.

I will evaluate the following option:
a) rewrite the binding certificate code to only use available classes
b) use the maven-shade-plugin as in the ``org.openhab.io.jetty.certificate`` bundle ``pom.xml`` to reduce the jar size

Reply:
From what I can tell, `org.openhab.io.jetty.certificate` doesn't export any of the `org.bouncycastle.*` classes that it has shaded internally. This means that you can't rely on `org.openhab.io.jetty.certificate` as a dependency to get those classes from.

Reply:
Are you having any trouble getting the binding to install?

Reply:
> This means that you can't rely on org.openhab.io.jetty.certificate as a dependency to get those classes from.

Ok, one reason more to focus option b) and shade bouncycastle jars on my one

Reply:
> Are you having any trouble getting the binding to install? Or does it install and just throw errors?

The trouble started after a successful compilation and bundle installation. It seems to me that all dependencies inside the OSGi container were resolved, because I was able to start the Bosch SHC Bridge Thing. But during the very first initialization I got exceptions during the keystore and the self signed certificate creation.

I will write unit tests to have a defined environment and reproduce steps for keystore and self signed certificate creation. Then I can evaluate different dependencies and code combinations.

Reply:
FYI: A first "copy & paste" of the maven-shade-plugin settings from the ``org.openhab.io.jetty.certificate`` [pom.xml](https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core.io.jetty.certificate/pom.xml#L28)  to this binding pom resulted in the error

    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.2.1:shade (default) on project org.openhab.binding.boschshc: Error creating shaded jar: Invalid signature file digest for Manifest main attributes -> [Help 1]

Reply:
Since you are going to shade the bouncycastle libraries into you binding anyway, the easiest way to do that would be to give those dependencies a `compile` scope in your pom file. Doing so would automatically get those dependencies shaded into your final binding jar. You also wouldn't need to list the dependencies in your feature.xml file either in that case.

Reply:
The current PR content has the ``compile`` scope for the bouncycastle dependencies and the shading results in a 3.4MB big  jar which is working.
It was requested to reduce the jar size and up to now I found no feasible way, because
a) the ``org.openhab.io.jetty.certificate`` jar doesn't provide necessary classes currently used in the ``BoschSslUtil`` class during runtime
b) I have no working maven pom to shade _and_ reduce the jar size during compile time

Reply:
When @Hilbrand made his suggestion I don't think he was aware that the `org.openhab.io.jetty.certificate` doesn't export the bouncycastle packages. So depending on the core to provide those dependencies isn't an option at the moment. For now I suggest leaving the dependencies embedded for now and once the core exposes the bouncycastle packages we can try to readdress the sizing issue.